elastic/kibana
# [Synthetics] Monitor Add/Edit - Fix form error states

Related to #154855

~Note: Should be merged post-GA, considering the magnitude of change.~

## Summary

- Connects form validation states with react-hoom-form's state (which was the reason of the reported bug that the error state was only reflected in euiForm and not the react-hook-form)
- Changes `reValidateMode` from `onChange` to `onSubmit` and manually invalidates the form when fields change so that dependent validations can be executed reactively e.g. fixes the following state:
  <img width="703" alt="Screenshot 2023-05-04 at 00 10 03" src="https://user-images.githubusercontent.com/2748376/236061192-a9417915-1eca-4cff-b871-b680955ec15a.png">
- Extracts validation and dependencies logic into a hook to simplify state management
- Fixes the following error where "Response body max bytes" field appear despite "Index response body" flag being unchecked.
  <img width="1205" alt="Screenshot 2023-05-05 at 19 35 27" src="https://user-images.githubusercontent.com/2748376/236527581-ff62550a-2679-431e-8571-da02a89c0a32.png">
- Fixes a case where updated values from for nested fields (such as "Response body contains JSON") weren't being passed to monitor inspect component ("Inspect configuration" flyout on monitor Add/Edit).
  
### Testing
- Since the core logic of the Monitor Add/Edit has been changed, the Add/Edit functionality should be thoroughly tested.
- Create at least one monitor of each type and ensure input validation works.
- Update above monitors and ensure everything works.
- Try to create a monitor with duplicate name and make sure the validation error is shown in time.
- Also test Getting Started page which appears while creating the first ever monitor in a Kibana space.
